library/src/main/java/com/novoda/downloadmanager/lib/DownloadDeleter.java

```diff
@@ -0,0 +1,72 @@
+package com.novoda.downloadmanager.lib;
+
+import android.content.ContentResolver;
+import android.content.ContentValues;
+import android.net.Uri;
+import android.text.TextUtils;
+
+import com.novoda.notils.logger.simple.Log;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Set;
+
+class DownloadDeleter {
+
+    private final ContentResolver resolver;
+
+    public DownloadDeleter(ContentResolver resolver) {
+        this.resolver = resolver;
+    }
+
+    public void deleteFileAndDatabaseRow(DownloadInfo info) {
+        deleteFileAndMediaReference(info);
+        resolver.delete(info.getAllDownloadsUri(), null, null);
+    }
+
+    public void deleteFileAndMediaReference(DownloadInfo info) {
```

Comment:
see the and in this method. That means internally it can look like this
```java
public void deleteFileAndMediaReference(DownloadInfo info) {
          deleteFile(info);
          deleteMediaReference(info);
}
```

FYI your method name should actually be `deleteMediaReferenceAndFile()`

If you had those two smaller methods, they could be easier to read with a *guard clause* & *prefer the positive*:

```java
public void deleteFileAndMediaReference(DownloadInfo info) {
     String mediaUri = info.mMediaProviderUri;
     if (TextUtils.isEmpty(mediaUri)) {
          return;
     }
     resolver.delete(Uri.parse(mediaUri), null, null);
}
```